Els fitxers .xml.patch són fitxers XML que indiquen qué corregir en un altre fitxer XML i com fer-ho.


=== AFEGIR (UN) ELEMENT(S) ===

Guardar sencer l'arbre per sobre l'/els element(s) i afegir-los-hi, com si es tractés d'una versió simplificada del fitxer per corregir.
No s'ha de guardar fins i tot els elements germans. Per defecte, els elements seran afegits al final, però és possible canviar això (mirar posar en ordre alfabètic a sota).


=== POSAR UN ELEMENT EN COMENTARI ===

Inserir la versió en comentari de l'element; el progamari l'actualitzarà.
Tots els elements corresponsals al lloc de l'arbre seràn actualitzats.


=== SORTIR UN ELEMENT D'UN COMENTARI ===

Inserir la versió fora de comentari de l'element; el programari l'actualitzarà.
Tots els elements corresponsals al lloc de l'arbre seràn actualitzats.


=== POSAR ELEMENTS EN ORDRE ALFABÈTIC ===

Afegir una etiqueta <sort>. No ha de ser al final, però ja que la classificació es fa sobre tots els elements després que els canvis siguin acabats, es recomana fer-ho.
PRECAUCIÓ: només fer servir aquesta funcionalitat quan tots els elements germans són de mateixa natura (tenen la mateixa etiqueta i la sva clau de classificació és disponible),
sinó es podria obtenir resultats estranys o el programari podria acabar anormalment.

<sort key="[key]" prepends="[prepends]"/>
   on:
[key] és el nom est le nom de l'atribut que conté el text per ser posat en ordre alfabètic. Pel text entre etiquetes, utilitzar "_text".
[prepends] és una llista de valors "key", separats per comes, d'un element que no ha de ser posat en ordre alfabètic i ha de ser anteposat en la llista.

Els comentaris que contenen un element germà seran també posats en ordre alfabètic com si aquest element no fos posat en comentari.
Els comentaris que no contenen un element germà seran anteposats.

Per exemple:

<?xml version="1.0" encoding="utf-8" standalone="yes"?>
<floater>
 <combo_box allow_text_entry="true" follows="left|top" height="23" left_pad="10" max_chars="135" mouse_opaque="true" name="translate_language_combo" top_delta="-5" width="190">
  <combo_box.item label="Català (Catalan)" name="Catalan" value="ca" />
  <combo_box.item label="Français canadien (French)" name="Canadian_French" value="fr_CA" />
  <sort key="label" prepends="System Default,English"/>
 </combo_box>
</floater>

El correctiu a dalt:

1) Afegira els «combo box item» "Català (Catalan)" i "Français canadien (French)";
2) Farà una llista dels «combo box item» "System Default" i "English", així com els comentaris que no contenen un «combo box item»;
3) Farà una llista en ordre alfabètic de tots els altres «combo box item» segons el seu atribut «label», incloient-hi aquells afegits i aquells en comentari.
